### PR TITLE
Ui/transform templates

### PR DESCRIPTION
**Pull request by chelshaw**
This PR enables CRUD actions for Transform templates in the UI. 

Built-in templates are not readable or editable, regardless of user's capabilities
readable-templates

You may create or update templates, but the name of the template is not editable.
create-show-edit

Deleting templates works
delete-template

User with list only permissions cannot click on templates to view
list-only-policy

### CODE REVIEW

File: ui/app/templates/components/transform-template-edit.hbs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    </div>
+  </div>
+</form>
+{{else}}
+<div class="box is-fullwidth is-sideless is-paddingless is-marginless">
+  {{#each model.attrs as |attr|}}
+    {{#if (eq attr.type "object")}}
+      {{info-table-row label=(capitalize (or attr.options.label (humanize (dasherize attr.name)))) value=(stringify (get model attr.name))}}
+    {{else if (eq attr.type "array")}}
+      {{info-table-row
```
Comment (Monkeychip) [inline ui/app/templates/components/transform-template-edit.hbs:112]: will these items in the array ever need to be linked?  If yes, then we can use the new info-table-item-array component.

File: ui/app/templates/partials/secret-list/transform-list-item.hbs [hunk 2]
```diff
@@ -1,77 +1,5 @@
...
-        {{if (eq item.id ' ') '(self)' (or item.keyWithoutParent item.id)}}
-      </div>
-    </div>
-  </div>
-{{/if}}
+<TransformListItem
+  @item={{item}}
+  @itemPath={{concat options.modelPrefix item.id}}
+  @itemType={{options.item}}
+/>
```
Comment (Monkeychip) [inline ui/app/templates/partials/secret-list/transform-list-item.hbs:5]: space at end of file.

File: ui/tests/integration/components/transform-list-item-test.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import EmberObject from '@ember/object';
```
Comment (Monkeychip) [inline ui/tests/integration/components/transform-list-item-test.js:1]: nice work on this. it's comforting to see all the capability checks in here.

**Review by Monkeychip — APPROVED** [review]
Just a couple of questions.  Nothing blocking.

File: ui/app/models/transform/template.js [hunk 4]
```diff
@@ -1,7 +1,56 @@
...
 
-export default DS.Model.extend({
-  name: DS.attr('string'),
-  alphabet: DS.belongsTo('transform/alphabet'),
-  transformations: DS.hasMany('transformation'),
+const { attr } = DS;
+
+const Model = DS.Model.extend({
+  useOpenAPI: false,
+  // getHelpUrl: function(backend) {
```
[the remark this answers is not in the document]
Comment (chelshaw) [reply ui/app/models/transform/template.js:11]: IT should be deleted. I was playing around with OpenAPI but forgot to remove 👍

File: ui/app/templates/components/transform-template-edit.hbs (hunk 1 above)
Comment (chelshaw) [reply ui/app/templates/components/transform-template-edit.hbs:112]: It already uses that component within `info-table-row` so I think the if check is just to make sure we pass the right props

**Pull request merged**